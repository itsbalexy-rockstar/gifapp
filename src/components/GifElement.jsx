import React from "react";
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import { CardActionArea } from '@mui/material';
import PropTypes from 'prop-types'


const GifElement = ({ title, url }) => {
  return (
    <>
      <Card sx={{ maxWidth: 345 }}>
        <CardActionArea>
          <CardMedia
            component="img"
            height="140"
            image={url}
            alt={title}
            className="img"
          />
          <CardContent>
            <Typography 
              gutterBottom variant="h5" 
              component="div" 
              className="title">
              {title}
            </Typography>
          </CardContent>
        </CardActionArea>
      </Card>
    </>
  );
};

GifElement.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

export default GifElement;
