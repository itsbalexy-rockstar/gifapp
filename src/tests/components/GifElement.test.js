import { shallow } from "enzyme"
import React from 'react'
import GifElement from "../../components/GifElement"

describe('Tests in <GifElement />', () => {

    const title = 'Un titulo'
    const image = 'https://localhost/algo.png'
    const wrapper = shallow(<GifElement title={title} url={image} />)

    test('Should show the correctly component', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('Should contains a title', () => {
        const paragraph = wrapper.find('.title')
        expect(paragraph.text().trim()).toBe(title)
    });
    test('Should contains a title and alt properties', () => {
        const gif = wrapper.find('.img')
        expect(gif.prop('image')).toBe(image)
        expect(gif.prop('alt')).toBe(title)
    });
    
})