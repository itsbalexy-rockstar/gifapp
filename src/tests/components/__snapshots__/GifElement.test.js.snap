// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifElement /> Should show the correctly component 1`] = `
<Fragment>
  <ForwardRef(Card)
    sx={
      Object {
        "maxWidth": 345,
      }
    }
  >
    <ForwardRef(CardActionArea)>
      <ForwardRef(CardMedia)
        alt="Un titulo"
        className="img"
        component="img"
        height="140"
        image="https://localhost/algo.png"
      />
      <ForwardRef(CardContent)>
        <ForwardRef(Typography)
          className="title"
          component="div"
          gutterBottom={true}
          variant="h5"
        >
          Un titulo
        </ForwardRef(Typography)>
      </ForwardRef(CardContent)>
    </ForwardRef(CardActionArea)>
  </ForwardRef(Card)>
</Fragment>
`;
